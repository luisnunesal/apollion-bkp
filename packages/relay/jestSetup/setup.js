const { JSDOM } = require('jsdom');
const server = require('./testServer');

module.exports = async () => {
  const httpServer = await server.listen(5000, () => {
    console.log(`\n test server running at: \n${JSON.stringify(httpServer.address(), null, 2)}`);
  });

  Object.assign(global, { httpServer });
};
