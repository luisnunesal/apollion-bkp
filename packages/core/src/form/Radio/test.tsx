import 'jest-styled-components';

import React from 'react';

import { Radio } from '.';

import { fireEvent, render } from '../../__tests__/setup';

describe.skip('Radio', () => {
  it('Should render with disabled={true}', () => {
    const { getByLabelText, container } = render(<Radio name="Input" label="Input" disabled />);

    const input = getByLabelText('Input') as HTMLInputElement;
    fireEvent.click(container);

    expect(container.firstChild).toMatchSnapshot();
    expect(input.checked).toBe(false);
  });

  it('Should match the layout', () => {
    const { container } = render(<Radio name="Input" label="Input" />);

    fireEvent.click(container);
    expect(container.firstChild).toMatchSnapshot();
  });
});
