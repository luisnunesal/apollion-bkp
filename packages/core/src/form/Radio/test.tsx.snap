// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio Should match the layout 1`] = `
.c0 {
  position: relative;
  text-rendering: optimizeLegibility;
  line-height: 1;
  margin: 0;
  padding: 0;
  color: #4F7F90;
  font-size: 0.9rem;
}

.c6 {
  width: auto;
  height: auto;
}

.c9 {
  fill: #003750;
}

.c3 {
  position: absolute;
  left: 0;
  top: 0;
  margin: 0;
  height: 100%;
  width: 100%;
  z-index: 1;
  opacity: 0;
  cursor: pointer;
}

.c5 {
  min-width: 22px;
  width: 22px;
  height: 22px;
  border-radius: 22px;
  margin-right: 0.75rem;
  border: 1px solid #143D66;
}

.c8 {
  width: 18px;
  height: 18px;
  opacity: 0;
  fill: #000000;
  -webkit-transform: translate(1.8px,1.8px);
  -ms-transform: translate(1.8px,1.8px);
  transform: translate(1.8px,1.8px);
}

.c1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 0rem 2rem 1rem 0rem;
  color: #143D66;
}

.c1 > .c4 {
  border: 1px solid #143D66;
}

.c1 > .c4 > .c7 > * {
  fill: #3399FF;
}

.c1:hover > .c4,
.c1:focus > .c4 {
  border-color: #297ACC;
}

.c1 > .c2:checked ~ .c4 {
  border-color: #3399FF;
}

.c1 > .c2:checked ~ .c4 > .c7 {
  opacity: 1;
}

<label
  class="c0 c1"
>
  <input
    class="c2 c3"
    name="Input"
    type="radio"
  />
  <div
    class="c4 c5"
  >
    <svg
      class="c6 c7 c8"
      height="22"
      viewBox="0 0 22 22"
      width="22"
    >
      <g
        class="c9"
      >
        <path
          class=""
          d="M 10 19 C 5.037380218505859 19 1 14.96261978149414 1 10 C 1 5.037380218505859 5.037380218505859 1 10 1 C 14.96261978149414 1 19 5.037380218505859 19 10 C 19 14.96261978149414 14.96261978149414 19 10 19 Z\\"/><path class=\\"d\\" d=\\"M 10 2 C 5.588789939880371 2 2 5.588789939880371 2 10 C 2 14.41121006011963 5.588789939880371 18 10 18 C 14.41121006011963 18 18 14.41121006011963 18 10 C 18 5.588789939880371 14.41121006011963 2 10 2 M 10 0 C 15.52285003662109 0 20 4.477149963378906 20 10 C 20 15.52285003662109 15.52285003662109 20 10 20 C 4.477149963378906 20 0 15.52285003662109 0 10 C 0 4.477149963378906 4.477149963378906 0 10 0 Z"
        />
      </g>
    </svg>
  </div>
  Input
</label>
`;

exports[`Radio Should render with disabled={true} 1`] = `
.c0 {
  position: relative;
  text-rendering: optimizeLegibility;
  line-height: 1;
  margin: 0;
  padding: 0;
  color: #4F7F90;
  font-size: 0.9rem;
}

.c6 {
  width: auto;
  height: auto;
}

.c9 {
  fill: #003750;
}

.c3 {
  position: absolute;
  left: 0;
  top: 0;
  margin: 0;
  height: 100%;
  width: 100%;
  z-index: 1;
  opacity: 0;
  cursor: pointer;
}

.c5 {
  min-width: 22px;
  width: 22px;
  height: 22px;
  border-radius: 22px;
  margin-right: 0.75rem;
  border: 1px solid #143D66;
}

.c8 {
  width: 18px;
  height: 18px;
  opacity: 0;
  fill: #000000;
  -webkit-transform: translate(1.8px,1.8px);
  -ms-transform: translate(1.8px,1.8px);
  transform: translate(1.8px,1.8px);
}

.c1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 0rem 2rem 1rem 0rem;
  color: #143D66;
  color: #BBCED4;
}

.c1 > .c4 {
  border: 1px solid #143D66;
}

.c1 > .c4 > .c7 > * {
  fill: #3399FF;
}

.c1:hover > .c4,
.c1:focus > .c4 {
  border-color: #297ACC;
}

.c1 > .c2:checked ~ .c4 {
  border-color: #3399FF;
}

.c1 > .c2:disabled {
  cursor: default;
}

.c1 > .c2:disabled ~ .c4 {
  background-color: #F1F5F6;
  border-color: #F1F5F6;
}

.c1 > .c2:disabled ~ .c4 > .c7 > * {
  fill: #BBCED4;
}

.c1 > .c2:checked ~ .c4 > .c7 {
  opacity: 1;
}

<label
  class="c0 c1"
  disabled=""
>
  <input
    class="c2 c3"
    disabled=""
    name="Input"
    type="radio"
  />
  <div
    class="c4 c5"
  >
    <svg
      class="c6 c7 c8"
      height="22"
      viewBox="0 0 22 22"
      width="22"
    >
      <g
        class="c9"
      >
        <path
          class=""
          d="M 10 19 C 5.037380218505859 19 1 14.96261978149414 1 10 C 1 5.037380218505859 5.037380218505859 1 10 1 C 14.96261978149414 1 19 5.037380218505859 19 10 C 19 14.96261978149414 14.96261978149414 19 10 19 Z\\"/><path class=\\"d\\" d=\\"M 10 2 C 5.588789939880371 2 2 5.588789939880371 2 10 C 2 14.41121006011963 5.588789939880371 18 10 18 C 14.41121006011963 18 18 14.41121006011963 18 10 C 18 5.588789939880371 14.41121006011963 2 10 2 M 10 0 C 15.52285003662109 0 20 4.477149963378906 20 10 C 20 15.52285003662109 15.52285003662109 20 10 20 C 4.477149963378906 20 0 15.52285003662109 0 10 C 0 4.477149963378906 4.477149963378906 0 10 0 Z"
        />
      </g>
    </svg>
  </div>
  Input
</label>
`;
